src/Analyzer/Resolve/IdentifierResolver.cpp

```diff
@@ -161,35 +161,29 @@ IdentifierResolveResult IdentifierResolver::tryResolveTableIdentifierFromDatabas
     storage_id = context->resolveStorageID(storage_id);
     bool is_temporary_table = storage_id.getDatabaseName() == DatabaseCatalog::TEMPORARY_DATABASE;
 
-    StoragePtr storage;
+    DatabaseAndTable database_and_table;
     TableLockHolder storage_lock;
+    std::optional<Exception> exception;
 
     if (is_temporary_table)
-        storage = DatabaseCatalog::instance().getTable(storage_id, context);
+        database_and_table = DatabaseCatalog::instance().getTableImpl(storage_id, context, &exception);
     else if (auto refresh_task = context->getRefreshSet().tryGetTaskForInnerTable(storage_id))
     {
         /// If table is the target of a refreshable materialized view, it needs additional
         /// synchronization to make sure we see all of the data (e.g. if refresh happened on another replica).
-        std::tie(storage, storage_lock) = refresh_task->getAndLockTargetTable(storage_id, context);
+        std::tie(database_and_table.second, storage_lock) = refresh_task->getAndLockTargetTable(storage_id, context);
     }
     else
-        storage = DatabaseCatalog::instance().tryGetTable(storage_id, context);
+        database_and_table = DatabaseCatalog::instance().getTableImpl(storage_id, context, &exception);
```

Comment:
The thing that worries me: 
Previously, we used tryGetTable, and it seems like the situation when the identifier was not resolved from the catalog was expected and could be handled later. But now, for every not-successful lookup, we create an exception object which collects stacktrace (which we will potentially not use).

I don't know how often this happens. But, just in case, maybe we get just a error code + error message from the DatabaseCatalog? It still would be an extra string copy, but maybe ok for this.

Reply:
>But now, for every not-successful lookup, we create an exception object which collects stacktrace (which we will potentially not use).

The exception will be created anyway

https://github.com/ClickHouse/ClickHouse/blob/e10665a7152755d4de5efc675a2e123a1154be14/src/Interpreters/DatabaseCatalog.cpp#L360

>I don't know how often this happens. But, just in case, maybe we get just a error code + error message from the DatabaseCatalog? It still would be an extra string copy, but maybe ok for this.

This is totally OK with me, but first, we I want to ensure that everybody is OK with this change.

And apparently @novikd don't like it, I don't have strong opinion about this patch, so let's close it for now, maybe later we will got some examples that will prove that it worth it